core/src/main/java/org/apache/calcite/adapter/enumerable/PhysType.java

```diff
@@ -117,6 +117,25 @@ Expression fieldReference(Expression expression, int field,
    */
   Expression generateAccessor(List<Integer> fields);
 
+  /** Similar to {@link #generateAccessor(List)}, but if one of the fields is <code>null</code>,
+   * it will return <code>null</code>.
+   *
+   * <p>For example:
+   *
+   * <blockquote><pre>
+   * new Function1&lt;Employee, Object[]&gt; {
+   *    public Object[] apply(Employee v1) {
+   *        return v1.&lt;fieldN&gt; == null
+   *            ? null
+   *            : v1.&lt;fieldM&gt; == null
+   *                ? null
+   *                : FlatLists.of(v1.&lt;fieldN&gt;, v1.&lt;fieldM&gt;);
+   *    }
+   * }
+   * }</pre></blockquote>
+   */
+  Expression generateNullableAccessor(List<Integer> fields);
```

Comment:
Why not modify the original function implement, but create a new function interface.

Reply:
This is a fair point. It could be done, but I think that the "old" method was still useful. Even though it would not be used any more within Calcite core, it might be used by some Calcite clients (and I didn't want to "break" their code by changing the implementation); so I considered it was arguably better (and probably safer) to keep the old method as it was, and add a new one with the peculiarities which are required for our hashJoin algorithm.

Reply:
If so, IMHO it's better to use a name like generateIgnoreNullAccessor. `generateAccessor` and `generateNullableAccessor` makes me feel like, if my fields is nullable I should use the last one.

Reply:
Yes, agree. I struggled to find a name for the new method. My initial thought was to name it "generateAccessorWithoutNulls" , which is a bit similar to the one you propose, wdyt?
However, I also had the impression that "generateAccessorWithoutNulls" (or "generateIgnoreNullAccessor" / "generateAccessorIgnoreNull") might give the impression that null values will be skipped and not returned by the accessor, and that is not the purpose of the method.
In any case, this is juts a detail that does not invalidate the fix, let's try to find the most suitable (or at least the less confusing) name for the new method, I'm totally open to suggestions (maybe some other reviewer will chime in with other proposals).